components/suggest/src/store.rs

```diff
@@ -49,6 +57,35 @@ pub struct SuggestStore {
     inner: SuggestStoreInner<remote_settings::Client>,
 }
 
+/// For records that aren't currently parsable,
+/// the record ID and the schema version it's first seen in
+/// is recorded in the meta table using `UNKNOWN_RECORDS_META_KEY` as its key.
+/// On the first ingest after an upgrade, re-request those records from Remote Settings,
+/// and try to ingest them again.
+#[derive(Deserialize, Serialize, Default, Debug)]
+#[serde(transparent)]
+pub struct UnparsableRecords(pub BTreeMap<String, UnparsableRecord>);
+
+impl FromSql for UnparsableRecords {
+    fn column_result(value: rusqlite::types::ValueRef<'_>) -> rusqlite::types::FromSqlResult<Self> {
+        serde_json::from_str(value.as_str()?)
+            .map_err(|err| rusqlite::types::FromSqlError::Other(Box::new(err)))
+    }
+}
+
+impl ToSql for UnparsableRecords {
+    fn to_sql(&self) -> rusqlite::Result<rusqlite::types::ToSqlOutput<'_>> {
+        Ok(ToSqlOutput::from(serde_json::to_string(self).map_err(
+            |err| rusqlite::Error::ToSqlConversionFailure(Box::new(err)),
+        )?))
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct UnparsableRecord {
+    pub(crate) schema_version: u32,
+}
```

Comment:
```suggestion
pub(crate) struct UnparsableRecord {
    pub schema_version: u32,
}
```

nitpick: I like to mark crate-internal types as `pub(crate)`, so that it's a bit easier to tell what's exposed to consumers, and what's part of the crate's implementation.

For fields and methods _within_ a `pub(crate)` type, though, I usually go with `pub`—Rust won't let `pub(crate)` types leak outside of the crate, so it's OK to use a more permissive (and shorter! 😊) visibility specifier for its fields.